### PR TITLE
feat(UI): Add, edit and delete widget

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This implements the form that adds and edits a widget. It also implement the delete widget feature.

https://github.com/centreon/centreon/assets/12515407/9e976ef0-4ecc-4fd6-ab98-3118caa14f1d

https://github.com/centreon/centreon/assets/12515407/7e85c307-f633-4e9b-8c46-c245b37bd53a

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

- Create a dashboard
- Add a widget
- -> The widget is added to the dashboard
- Edit the created widget
- -> The widget is correctly edited
- Delete the widget
- -> The widget is removed

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -17904,6 +17904,39 @@ msgstr "Erreur lors de la modification partielle d'une configuration de notifica
...
+msgid "No widget found"
+msgstr "Aucun widget trouvé"
+
+msgid "Delete a widget"
+msgstr "Supprimer un widget"
+
+msgid "Do you want to delete this widget?"
+msgstr "Voulez-vous supprimer ce widget ?"
+
+msgid "Widgets library"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17919]:
```suggestion
msgid "Widget library"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -17904,6 +17904,39 @@ msgstr "Erreur lors de la modification partielle d'une configuration de notifica
...
+msgstr "Bibliothèque de widgets"
+
+msgid "Please choose a widget to activate the preview"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17922]:
```suggestion
msgid "Select a widget type to activate the preview"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -17904,6 +17904,39 @@ msgstr "Erreur lors de la modification partielle d'une configuration de notifica
...
+msgstr "Veuillez choisir un widget pour activer l'aperçu"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17923]:
```suggestion
msgstr "Sélectionnez un type de widget pour activer l'aperçu"
```

File: centreon/www/front_src/src/Dashboards/Dashboard/Dashboard.cypress.spec.tsx [hunk 4]
```diff
@@ -195,199 +240,105 @@ const initializeAndMount = ({
...
-      });
   });
 
-  it('saves the dashboard when the corresponding button is clicked and the dashboard is changed', () => {
-    initializeBlocker();
-    const store = initializeAndMount({});
-
-    cy.waitForRequest('@getDashboardDetails');
+  describe('Edit widget', () => {
+    it('edits a widget when the corresponding button is clicked, the widget type is changed the edit button is clicked', () => {
```
Comment (cg-tw) [inline centreon/www/front_src/src/Dashboards/Dashboard/Dashboard.cypress.spec.tsx:303]:
```suggestion
    it('edits a widget when the corresponding button is clicked, the widget type is changed and the edit button is clicked', () => {
```

File: centreon/www/front_src/src/Dashboards/Dashboard/translatedLabels.ts [hunk 5]
```diff
@@ -11,3 +11,21 @@ export const labelQuitDashboardChangesNotSaved =
   'If you leave {{dashboardName}}, your changes will not be saved.';
 export const labelAdd = 'Add';
 export const labelYourDashboardHasBeenSaved = 'Your dashboard has been saved';
+export const labelSelectAWidgetType = 'Select a widget type';
+export const labelSearch = 'Search';
+export const labelNoWidgetFound = 'No widget found';
+export const labelDeleteAWidget = 'Delete a widget';
+export const labelDoYouWantToDeleteThisWidget =
+  'Do you want to delete this widget?';
+export const labelWidgetsLibrary = 'Widgets library';
```
Comment (cg-tw) [inline centreon/www/front_src/src/Dashboards/Dashboard/translatedLabels.ts:20]:
```suggestion
export const labelWidgetsLibrary = 'Widget library';
```

**Comment by bdauria** [comment]
@Thebarda the action icons on the top right hand side of the widget are too close from the border. I'dd add a little bit of spacing. cc @Wedierre

**Comment by Thebarda** [comment]
> @Thebarda the action icons on the top right hand side of the widget are too close from the border. I'dd add a little bit of spacing. cc @Wedierre

I've added a small horizontal spacing between icons and increased the margin-right but the header will be reworked later in order to integrate the name of the widget (such as Grafana)

**Review by bdauria — APPROVED** [review]

**Review by Noha-ElAbrouki — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by wtermellil — APPROVED** [review]

File: centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -15751,6 +15751,39 @@ msgstr ""
...
+# msgid "No widget found"
+# msgstr ""
+
+# msgid "Delete a widget"
+# msgstr ""
+
+# msgid "Do you want to delete this widget?"
+# msgstr ""
+
+# msgid "Widget library"
```
Comment (smau-centreon) [inline centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po:15766]:
```suggestion
# msgid "Widgets library"
```

File: centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -16086,6 +16086,39 @@ msgstr ""
...
+# msgid "No widget found"
+# msgstr ""
+
+# msgid "Delete a widget"
+# msgstr ""
+
+# msgid "Do you want to delete this widget?"
+# msgstr ""
+
+# msgid "Widget library"
```
Comment (smau-centreon) [inline centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po:16101]:
```suggestion
# msgid "Widgets library"
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -16550,6 +16550,39 @@ msgstr ""
...
+# msgid "No widget found"
+# msgstr ""
+
+# msgid "Delete a widget"
+# msgstr ""
+
+# msgid "Do you want to delete this widget?"
+# msgstr ""
+
+# msgid "Widget library"
```
Comment (smau-centreon) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:16565]:
```suggestion
# msgid "Widgets library"
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -16539,6 +16539,39 @@ msgstr "Você não salvou as alterações, deseja continuar?"
...
+# msgid "No widget found"
+# msgstr ""
+
+# msgid "Delete a widget"
+# msgstr ""
+
+# msgid "Do you want to delete this widget?"
+# msgstr ""
+
+# msgid "Widget library"
```
Comment (smau-centreon) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:16554]:
```suggestion
# msgid "Widgets library"
```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

File: centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po (hunk 6 above)
Comment (Thebarda) [reply centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po:15766]: Is it "widgets" or "widget" because @cg-tw told me to remove the "s" from "widgets"

Comment (cg-tw) [reply centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po:15766]: it is widget

Comment (smau-centreon) [reply centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po:15766]: ok!

**Review by smau-centreon — APPROVED** [review]

**Pull request merged**